File: shared-types/src/models/Board.ts
import { UserResponse } from './User';

export interface IBoard {
  _id: string;
  title: string;
  description?: string;
  owner: string | UserResponse;
  members: (string | UserResponse)[];
  columns: string[];
  createdAt: Date;
  updatedAt: Date;
}

export interface BoardResponse {
  _id: string;
  title: string;
  description?: string;
  owner: UserResponse;
  members: UserResponse[];
  columns: string[];
  createdAt: Date;
  updatedAt: Date;
}

export interface BoardListResponse {
  success: boolean;
  data: BoardResponse[];
  count: number;
}

File: backend-fastapi/app/models/board.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class UserResponse(BaseModel):
    id: str = Field(alias='_id')
    email: str
    username: str

    class Config:
        populate_by_name = True


class BoardResponse(BaseModel):
    id: str = Field(alias='_id')
    title: str
    description: Optional[str] = None
    owner: UserResponse
    members: list[UserResponse]
    columns: list[str]
    createdAt: datetime
    updatedAt: datetime

    class Config:
        populate_by_name = True


class BoardListResponse(BaseModel):
    success: bool
    data: list[BoardResponse]
    count: int
